br/pkg/task/restore.go

```diff
@@ -907,7 +907,7 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		if cfg.WithSysTable {
 			client.InitFullClusterRestore(cfg.ExplicitFilter)
 		}
-	} else if checkpointFirstRun && cfg.CheckRequirements {
+	} else if client.IsFull() && checkpointFirstRun && cfg.CheckRequirements {
```

Comment:
so, only full restore will check dup? is it expected? i think the proper solution is to check dup at ddl execution time no matter what kind of restore. @Leavrth @3pointer   WDYT?